symfony/src/Repository/EventRepository.php

```diff
@@ -51,4 +53,19 @@ public function getRecent(array $types = []): array
             ->enableResultCache(3600)
             ->getResult();
     }
+
+    public function getLatestEventTimestamp(): ?DateTimeImmutable
+    {
+        $resultData = $this
+            ->createQueryBuilder('e')
+            ->select('MAX(e.timestamp)')
+            ->getQuery()
+            ->enableResultCache(3600)
+            ->getSingleScalarResult();
```

Comment:
Since `null` could be a expected allowed scalar result, Doctrine throws `NoResultException`, which along with `NonUniqueResultException` inherits from `UnexpectedResultException`. Please catch `UnexpectedResultException` and return `null` in that case.

Reply:
Oh, thank you for the catch!!

Reply:
I wanted to investigate this so I could familiarize myself with PHP/the codebase more and figure out how to spot this bug myself in the future.

I checked out the source code for **doctrine/orm** and I think it isn't consistent in throwing this error. [`getSingleScalarResult`](https://github.com/doctrine/orm/blob/9c560713925ac5859342e6ff370c4c997acf2fd4/src/AbstractQuery.php#L797) does say it throws `NoResultException` if there's no result. But it only throws the error [if `hydrationMode` isn't `HYDRATE_SINGLE_SCALAR`](https://github.com/doctrine/orm/blob/9c560713925ac5859342e6ff370c4c997acf2fd4/src/AbstractQuery.php#L775). `executeIgnoreQueryCache` (if cache disabled or there's a cache miss) calls [`setHydrationMode`](https://github.com/doctrine/orm/blob/9c560713925ac5859342e6ff370c4c997acf2fd4/src/AbstractQuery.php#L905), but if the query is cached and there's a cache hit, then we don't call `setHydrationMode`. As a result, even though the documentation says it will throw the error, it doesn't throw if the query isn't cached.

I did this research because when I tried to cause the error intentionally (to verify it), I couldn't — `getSingleScalarResult` returned `null` when I had no rows in `EVENTS`.

I was also surprised to learn that even though I'd used `enableResultCache(3600)`, the query wasn't "cached" — it only ever called `executeIgnoreQueryCache`. You would need to call `setCacheable(true)` — but also, it seems the query cache isn't enabled?

**REGARDLESS!** I can't tell if this is intended behavior (documentation is wrong) or not (documentation is right and this is a bug). It's also what you want, so I've added the `try` statement. But I wanted to share this result because it seemed interesting?

Reply:
Cache:
- I am slowly mowing towards cache only on the controller/service level, the DB-level cache caused a big impact on memory; I was using two-level cache, this does not make much sense now that I think about it.
- It could be that the dev environment is disabling the cache whatsoever (for Doctrine, this possibly enables cache on prod: https://github.com/veelkoov/fuzzrake/blob/3e5132c9a1995ecb01891bafffd6e58a48fcf8f4/symfony/config/packages/doctrine.yaml#L40 ; I am unable to quickly find the settings for the framework)

Reply:
Doctrine Behavior: weird... I created a test here: https://github.com/veelkoov/fuzzrake/commit/a1677f3aee6fbfa5a6c61ad802dcb8dacc988a71 (branch [testing/doctrine-behavior-test](https://github.com/veelkoov/fuzzrake/compare/testing/doctrine-behavior-test) and you can verify the results by `./toolbox pu --filter testGetSingleScalarResult`. Could you try running the test in your environment? I've never used Doctrine outside Symfony.

Reply:
Now I get it. You are doing a `SELECT MAX()` which returns null if there are no rows. I am sorry, I didn't catch that earlier. So now let me think what will happen if a null will get returned....

Reply:
FWIW after your initial comment, I tested this code after deleting all of the rows from the DB. It returned `null` without throwing an error (which was why I explored the code, since the documentation says that it should throw an error).